src/lib/videotagging_extensions/detection.js

```diff
@@ -0,0 +1,193 @@
+const DetectionAlgorithmManager = require('../detection_algorithms').DetectionAlgorithmManager;
+
+function Detection(videotagging, visitedFrames) {
+    this.videotagging = videotagging;
+    this.visitedFrames = visitedFrames;
+    this.detectionAlgorithmManager = new DetectionAlgorithmManager();
+
+    var self = this;
+
+    //maps every frame in the video to an imageCanvas NOTE mapVideo clears the oncanplay eventListener
+    this.mapVideo = function(frameHandler, exportUntil) {
+        return new Promise((resolve, reject) => {
+            //init canvas buffer
+            var frameCanvas = document.createElement("canvas");
+            frameCanvas.width = self.videotagging.video.videoWidth;
+            frameCanvas.height = self.videotagging.video.videoHeight;
+            var canvasContext = frameCanvas.getContext("2d");
+
+            // start exporting frames using the canplay eventListener
+            self.videotagging.video.oncanplay = iterateFrames;
+            self.videotagging.video.currentTime = 0;
+            self.videotagging.playingCallback();
+
+            function iterateFrames() {
+                var frameId = self.videotagging.getCurrentFrame();
+                var isLastFrame;
+
+                switch(exportUntil) {
```

Comment:
Theoretically you can do here something more elegant instead of calling the "switch" for each frame. You can resolve the type of the "exportUntil" before you start the process and pass function that calculates the "isLastFrame". You can then use that function each time..can probably reduce the code.
You can define an object with the exportUntil types as key and functions as values and then just get the function according to that value...and init that object at the beginning of the module, etc.